core/src/main/java/com/linecorp/armeria/client/limit/DefaultConcurrencyLimit.java

```diff
@@ -65,7 +75,17 @@ int acquiredPermits() {
 
     @VisibleForTesting
     int availablePermits() {
-        return maxConcurrency - acquiredPermits.get();
+        int availablePermitCount = maxConcurrency() - acquiredPermits.get();
+        return Math.max(availablePermitCount, 0);
+    }
+
+    @VisibleForTesting
+    int maxConcurrency() {
+        int maxConcurrency = this.maxConcurrency.get();
+        if (maxConcurrency <= 0) {
```

Comment:
I guess we need to add a description that the `Limit` does not produce a negative value.